src/bson_mapper/test/stdx_optional_archiver_test.cpp

```diff
@@ -0,0 +1,299 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "catch.hpp"
+
+#include <bson_mapper/bson_archiver.hpp>
+#include <bson_mapper/stdx/optional.hpp>
+#include <bsoncxx/json.hpp>
+#include <cereal/types/vector.hpp>
+#include <fstream>
+
+using bson_mapper::stdx::optional;
+using bson_mapper::stdx::make_optional;
+using bson_mapper::stdx::nullopt;
+
+// Optional values.
+struct OptDataA {
```

Comment:
I opted not to inherit from the model because the tests in `bson_mapper/test/stdx_optional_archiver_test.cpp` are supposed to be independent from the `mongo_odm` namespace. If you look at `mongo_odm/test/model.cpp`, you'll see that I tested optional fields in a class that inherits from the model.